src/role_nextroomer.js

```diff
@@ -308,11 +308,15 @@ roles.nextroomer.settle = function(creep) {
     methods.push(Creep.constructTask);
   }
 
+  // Fill energy after building is done
+  if (creep.room.controller.level >= 4) {
+    methods.push(Creep.transferEnergy);
+  }
+
   if (creep.room.controller.level < 8) {
+
```

Comment:
Is this really what we want? Personally i'd prefer if they used their valuable work parts for upgrading.